src/blockdata/script.rs

```diff
@@ -449,13 +449,20 @@ impl Script {
     }
 
     #[cfg(feature="bitcoinconsensus")]
-    /// verify spend of an input script
-    /// # Parameters
-    ///  * index - the input index in spending which is spending this transaction
-    ///  * amount - the amount this script guards
-    ///  * spending - the transaction that attempts to spend the output holding this script
+    /// Shorthand for [Self::verify_with_flags] with flag [bitcoinconsensus::VERIFY_ALL]
     pub fn verify (&self, index: usize, amount: u64, spending: &[u8]) -> Result<(), Error> {
```

Comment:
Shouldn't this be `Amount`? (especially once we adopt wider `Amount` use like in #599)

Reply:
I prefer to keep this non API breaking and not change yet the `verify` method signature, but I changed the amount type in `verify_with_flags` in 69117a1f63231e8ee9821dfcecafafd7d8358774